firecracker-control/local.go

```diff
@@ -321,3 +324,26 @@ func (s *local) newShim(ns, vmID, containerdAddress string, shimSocket *net.Unix
 func isEADDRINUSE(err error) bool {
 	return err != nil && strings.Contains(err.Error(), "address already in use")
 }
+
+// validateID checks ID is valid and safe to use as a part of a file system path
+func validateID(id string) error {
+	const (
+		maxIDLen          = 255 // Maximum filename length is 255 characters on ext4
+		invalidCharacters = "\t\n\\/.:?* "
```

Comment:
So I don't think most of those are actually invalid.  But I'd like to propose an alternate approach here of defining a limited, valid character set rather than an invalid character set.  Something like this:  `[A-Za-z9-0-_]`.

Reply:
We could also consider re-using containerd's code for identifier validation: https://github.com/containerd/containerd/blob/6ba92a7921c558012c9ca87f2ca82432a683a527/identifiers/validate.go

Reply:
I'd be comfortable with containerd's validator as well.

Reply:
That's cool. I didn't notice containerd has it. Done.